src/gatsby-utils.ts

```diff
@@ -0,0 +1,113 @@
+import path from 'path';
+import glob from 'glob';
+import normalize from 'normalize-path';
+import {
+  graphql,
+  getIntrospectionQuery,
+  buildClientSchema,
+  GraphQLSchema,
+  IntrospectionQuery,
+} from 'graphql';
+
+import { Store } from 'gatsby';
+import getGatsbyDependents from 'gatsby/dist/utils/gatsby-dependents';
+
+const introspectionQuery = getIntrospectionQuery({
+  descriptions: false,
+});
+
+export type GatsbyTheme = {
+  themeDir: string,
+  [key: string]: any,
+};
+
+export type GatsbyPlugin = {
+  pluginFilepath: string,
+  [key: string]: any,
+};
+
+export function resolveThemes(themes: GatsbyTheme[]) {
+  return themes.reduce((merged, theme) => {
+    return [...merged, theme.themeDir];
+  }, [] as string[]);
+};
+
+interface LoadGatsbySourcesFn {
+  (props: {
+    store: Store,
+  }): Promise<string[]>;
+}
+// Original implementation from https://github.com/gatsbyjs/gatsby/blob/561d33e2e4/packages/gatsby/src/query/query-compiler.js
+export const loadGatsbyFiles: LoadGatsbySourcesFn = async ({
```

Comment:
I was wondering — why is this function necessary vs. using loadDocuments via @graphql-toolkit?

Reply:
It's becuase the behavior more closer to Gatsby's internal. The plugin goal is codegen for only gatsby so using Gatsby's internal makes sense to ensure documents are actually used by Gatsby.

Reply:
Removed this util.